chat_shell/chat_shell/api/v1/response.py

```
@@ -173,315 +208,96 @@ async def _stream_response(
-        event_type, data = await emitter.done(
-            content=full_content,
-            usage=usage_dict,
-            stop_reason="silent_exit" if is_silent_exit else "end_turn",
-            sources=formatted_sources,
-            silent_exit=is_silent_exit if is_silent_exit else None,
-            silent_exit_reason=silent_exit_reason if silent_exit_reason else None,
-            **done_extra_fields,
-        )
+            # Check if chat task completed with error
+            if chat_task.done() and chat_task.exception():
+                raise chat_task.exception()
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)